packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -10,6 +10,597 @@ const ScopeSet = require('../../../fxa-shared').oauth.scopes;
 const validators = require('./validators');
 const { metadataFromPlan } = require('./utils/subscriptions');
 
+const stripe = require('../payments/stripe');
+
+const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
+  'https://identity.mozilla.com/account/subscriptions';
+
+/** @typedef {import('stripe').Stripe.Invoice} Invoice */
+/** @typedef {import('stripe').Stripe.PaymentIntent} PaymentIntent */
+
+async function handleAuth(db, auth, fetchEmail = false) {
+  const scope = ScopeSet.fromArray(auth.credentials.scope);
+  if (!scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
+    throw error.invalidScopes('Invalid authentication scope in token');
+  }
+  const { user: uid } = auth.credentials;
+  let email;
+  if (!fetchEmail) {
+    ({ email } = auth.credentials);
+  } else {
+    const account = await db.account(uid);
+    ({ email } = account.primaryEmail);
+  }
+  return { uid, email };
+}
```

Comment:
As a caller, I wouldn't know what to do with the `fetchEmail` flag.  I'd get an email address if I set it to true, but I may or may not get an email address if it's `false`?  Or I'd always get an email address but the source of it is based on the flag?

Reply:
This was retained as-is, its just being moved. If you include it, then the email is fetched from the db, vs. extracted from the credentials. An email will always be included, its just how its retrieved. I'm not entirely clear on why it works like this as the function is from a previous PR.

Reply:
FWIW, this is [an existing util function that got moved](https://github.com/mozilla/fxa/pull/3917/files/0f875831223bfdc39bd92bb6c010511c3cf27007#diff-eedc9e44d94c523680530e9c145c9496L41). I think I wrote the original but I can't quite remember why it works the way it does.

I want to say I found a problem where `auth.credentials` almost never contains `email`, and so `fetchEmail` is true for those routes that absolutely need an email address. We probably could / should remove the bit that checks `auth.credentials`

Reply:
More context, since it bothers me not remembering why it works this way... But I think the context is that this is an issue from back in May that I vaguely meant to get back to eventually 😬 

https://github.com/mozilla/fxa/pull/1175#pullrequestreview-240147690